eventq_rabbitmq/lib/eventq_rabbitmq/rabbitmq_eventq_client.rb

```diff
@@ -13,10 +13,28 @@ def initialize(options={})
         @event_raised_exchange = EventRaisedExchange.new
         @serialization_manager = EventQ::SerializationProviders::Manager.new
         @signature_manager = EventQ::SignatureProviders::Manager.new
+
+        #this array is used to record known event types
+        @known_event_types = []
+      end
+
+      def registered?(event_type)
+        @known_event_types.include?(event_type)
+      end
+
+      def register_event(event_type)
+        if registered?(event_type)
+          return true
+        end
+
+        @known_event_types << event_type
+        true
       end
 
       def raise_event(event_type, event)
 
+        register_event(event_type)
```

Comment:
I have intentionally not added any check for this into the specs, as I am assuming that as part of the rabbitmq work for raising messages direct to a queue with delay you will be doing similar spec refactoring to separate integration specs from class logic? if so I will wait and add a check for registration into the `raise_event` method spec.